### PR TITLE
First draft of MovieList 

### PR DESCRIPTION
**Pull request by HelkaPL**
It's a first draft. 
It's still need some code fixind. 
But I need a break tooday ( I feel like i'm burning )
But there is some work done.

### CODE REVIEW

File: src/feautures/MovieList/MovieTile/Rates/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+import { StyledRates, Votes, StyledStar, Score } from "./styled";
+// i know ther can be a empty rating but i fixed it latter ( idt how its look's as empty)
+const Rates = ({ score = "7,8", votes = "35" }) => (
+    <StyledRates>
+        <StyledStar />
+        <Score>{score}</Score>
+        <Votes>{votes} vote{votes === "1" ? "" : "s"}</Votes>
```
Comment (mikolajgraczyk) [inline src/feautures/MovieList/MovieTile/Rates/index.js:8]: Good job here

File: src/feautures/MovieList/MovieTile/Rates/styled.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import styled from "styled-components";
+import { ReactComponent as Star } from "./star.svg";
+
+export const StyledRates = styled.div`
+    margin: 8px 0;
```
Comment (mikolajgraczyk) [inline src/feautures/MovieList/MovieTile/Rates/styled.js]: I would change margin to: `margin-top:  39px;` referring to figma

File: src/feautures/MovieList/styled.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import styled from "styled-components";
+
+export const StyledMovies = styled.div`
+    display: grid;
+    grid-template-columns: auto auto auto auto;
+    grid-gap: 24px;
+    margin-bottom: 40px;
+    `;
```
Comment (mikolajgraczyk) [inline src/feautures/MovieList/styled.js]: Don't forget about Media queries ;)

File: src/feautures/Search/Spinner/styled.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    animation: rotation 5s linear infinite;
+    @keyframes rotation {
+        0% {
+            transform: rotate(0deg);
+        }
+        100% {
+            transform: rotate(360deg);
+        }
+    }
+`;
```
Comment (mikolajgraczyk) [inline src/feautures/Search/Spinner/styled.js:17]: It's simple and it works, good job!

**Review by mikolajgraczyk — CHANGES_REQUESTED** [review]
Amazing work, I don't see many things to change here

File: src/feautures/MovieList/MovieTile/Rates/styled.js (hunk 2 above)
Comment (HelkaPL) [reply src/feautures/MovieList/MovieTile/Rates/styled.js]: There is no such think in figma... 
But maybe you got a point and better to stick rates to bottom border

Comment (mikolajgraczyk) [reply src/feautures/MovieList/MovieTile/Rates/styled.js]: obraz

That is the margin I talk about 😉

Comment (HelkaPL) [reply src/feautures/MovieList/MovieTile/Rates/styled.js]: Check again in diferent tile where there is no tags, or subtitle.. 
So its be better to stick it to bottom, and left 4px margin as it is for media..

File: src/feautures/MovieList/styled.js (hunk 3 above)
Comment (HelkaPL) [reply src/feautures/MovieList/styled.js]: I thinking about change to flex...

File: src/feautures/MovieList/MovieTile/Rates/styled.js (hunk 2 above)
Comment (mikolajgraczyk) [reply src/feautures/MovieList/MovieTile/Rates/styled.js]: You're right. It can be different above so sticking to the bottom is better option

File: src/feautures/MovieList/styled.js (hunk 3 above)
Comment (mikolajgraczyk) [reply src/feautures/MovieList/styled.js]: Why??

File: src/common/Section/index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { StyledSection, Title } from "./styled";
+
+const Section = ({ title, body }) => {
+
+    return (
+        <StyledSection>
+            <Title>{title}</Title>
```
Comment (marcin10lw) [inline src/common/Section/index.js]: maybe let's call it SectionTitle for code consistency

File: src/common/Section/styled.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import styled from "styled-components";
+
+const bpMobile = ({ theme }) => theme.breakpoint.mobile;
+
+export const StyledSection = styled.div`
```
Comment (marcin10lw) [inline src/common/Section/styled.js]: StyledSection should be section not div

File: src/common/Section/styled.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import styled from "styled-components";
+
+const bpMobile = ({ theme }) => theme.breakpoint.mobile;
```
Comment (marcin10lw) [inline src/common/Section/styled.js:3]: Nice you used it <3 !

File: src/common/Section/styled.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+export const StyledSection = styled.div`
+    margin: 0 auto;
+    padding: 56px 16px 0 16px;
+    max-width: 1368px;
```
Comment (marcin10lw) [inline src/common/Section/styled.js:7]: We will have to use `margin: 0 auto;` and `max-width: 1368px;` a lot in our project. My suggestion is to create Container common component with this styles

File: src/common/Section/styled.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+   
+    @media (max-width: ${bpMobile}px) {
+        padding: 24px 16px 0 16px;
+        max-width: 640px;
+  }
+`;
+
+export const Title = styled.h1`
```
Comment (marcin10lw) [inline src/common/Section/styled.js]: We will use Section more than once on a page so maybe change that to h2
image

File: src/common/Section/styled.js [hunk 10]
```diff
@@ -9,18 +9,22 @@ export const StyledSection = styled.div`
    
     @media (max-width: ${bpMobile}px) {
         padding: 24px 16px 0 16px;
+        max-width: 640px;
   }
 `;
 
 export const Title = styled.h1`
     margin: 0 0 24px 0;
+    width: 100vh;
```
Comment (mikolajgraczyk) [inline src/common/Section/styled.js]: I think you should remove this line. It causes trouble on narrow screens:
obraz

File: src/feautures/MovieList/MovieTile/index.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import Rates from "./Rates";
+import { MovieInfo, Poster, StyledTile, Subtitle, Title, Dummy } from "./styled";
+import Tags from "./Tags";
+
+
+export const MovieTile = ({ poster, title, subtitle, rate, votes }) => {
+
+    return (
+        <>
+            <StyledTile>
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/index.js]: Let's name it StyledMovieTile 😉

File: src/feautures/MovieList/styled.js [hunk 12]
```diff
@@ -3,6 +3,15 @@ import styled from "styled-components";
...
+    margin: 0 auto 40px auto;
+
+    @media (max-width: 632px) {
+        grid-template-columns: auto;
+    };
+
+    @media (min-width: 633px) and (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
+        grid-template-columns: auto auto;
+    };
+`;
```
Comment (mikolajgraczyk) [inline src/feautures/MovieList/styled.js]: I think one more media query is required:
obraz

File: src/feautures/MovieList/MovieTile/styled.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
+import styled from "styled-components";
+import { ReactComponent as DummyMovie } from "./video.svg";
+
+export const StyledTile = styled.div`
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js]: I think we should use react-router's Link component here:
`export const StyledTile = styled(Link)`

File: src/feautures/MovieList/MovieTile/styled.js [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    display: grid;
+    grid-template-columns: 1fr ;
+    grid-gap: 16px;
+    padding: 16px;
+    background: #FFF;
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js:9]: Maybe we should use white color from theme.js here

**Review by mikolajgraczyk — CHANGES_REQUESTED** [review]

File: src/feautures/MovieList/MovieTile/styled.js [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+    border-radius: 5px;
+    box-shadow: 0px 4px 12px rgba(186, 199, 213, 0.5);
+    width: 324px;
+    height: 650px;
+
+    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js:15]: Great work here man!

File: src/feautures/MovieList/MovieTile/index.js [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+                <Poster >
+                    {poster || <Dummy />}
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/index.js:12]: Our images from api response will be links so let's make Poster an img.
I's also suggest using ternary operator like this:
`{poster ? <Poster src={poster} alt={title} /> : <Dummy />}`

File: src/feautures/MovieList/MovieTile/styled.js [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+`;
+
+export const MovieInfo = styled.div`
+    position: relative;
+    margin: 0;
+    padding: 0;
+    height: 168px;
+`;
+
+export const Poster = styled.div`
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js:29]: I think Poster should be an image because we'll be getting image link from api res.
`display: flex;` is redundant I think

File: src/feautures/MovieList/MovieTile/styled.js [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 292px;
+    height: 434px;
+    background-color: #C4C4C4;
+    border-radius: 5px;
+
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js:38]: Use `object-fit: cover;` to make sure our image won't stretch weirdly 😉

File: src/feautures/MovieList/MovieTile/styled.js [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+   
+    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
+        width: 114px;
+        height: 169px;
+    }
+
+
+`;
+
+export const Title = styled.h2`
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js]: If we make our Section title h2 we should make this one h3 :)

File: src/feautures/MovieList/MovieTile/styled.js [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+    margin: 0;
+    color: #18181B;
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js]: Let's use theme.js color here

File: src/feautures/MovieList/MovieTile/styled.js [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+    font-family: 'Poppins';
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js]: We won't need that with font-family added to GlobalStyle.js

File: src/feautures/MovieList/MovieTile/styled.js [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+    line-height: 130%;
+
+    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
+        font-size: 16px;
+    }     
+`;
+
+export const Subtitle = styled.p`
+    margin: 8px 0;
+    color: #7e839a;
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js:62]: Let's use theme.js color here

File: src/feautures/MovieList/MovieTile/styled.js [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+    font-size: 16px;
+    line-height: 150%;
+
+    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
+        font-size: 13px;
+        line-height: 130%;
+    }       
+`;
+
+export const Dummy = styled(DummyMovie)`
```
Comment (marcin10lw) [inline src/feautures/MovieList/MovieTile/styled.js:73]: I think our Dummy should have same height, width and border-radius as Poster as we'll conditionally render them.
We can also use background-image to place our video.svg inside
I think it will make the code a bit simpler 💪

File: src/feautures/MovieList/index.js [hunk 24]
```diff
@@ -0,0 +1,68 @@
+import { MovieTile } from "./MovieTile";
+import { StyledMovies } from "./styled";
+
+const MovieList = () => {
+
+    return (
+        <StyledMovies>
```
Comment (marcin10lw) [inline src/feautures/MovieList/index.js]: Let's name it to StyledMovieList according to our convention :)

File: src/feautures/MovieList/styled.js [hunk 25]
```diff
@@ -0,0 +1,17 @@
+import styled from "styled-components";
+
+export const StyledMovies = styled.div`
```
Comment (marcin10lw) [inline src/feautures/MovieList/styled.js]: I think this should be ul

File: src/feautures/MovieList/styled.js [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+    display: grid;
+    grid-template-columns: auto auto auto auto;
```
Comment (marcin10lw) [inline src/feautures/MovieList/styled.js]: try using:

- `grid-template-columns: repeat(auto-fill, 324px);`
- `justify-content: center;`

For it to work make sure you set Section's side paddings to 0 💪

File: src/feautures/MovieList/styled.js (hunk 12 above)
Comment (HelkaPL) [reply src/feautures/MovieList/styled.js]: Fixed :)

**Review by HelkaPL — COMMENTED** [review]
Fixed i hope.

**Review by marcin10lw — CHANGES_REQUESTED** [review]
Nice job Kuba!

1. I think we should create movies and actors folders in features folder
2. We will use `margin: 0 auto;`, `max-width: 1368px;` a lot so i suggest to create Container in common folder for that
3. Same with unordered list:

- in common let's create a separate component TileList or something and give it this styles:
image
- then use it in our MovieList like that:
image
- If 4 movie columns won't appear try to find if some side padding doesn't disturb that

4. I think Pagination should not be a part of Section

In conclusion great work! We need some tweeks and we'll be good! 💪

File: src/common/Section/styled.js (hunk 8 above)
Comment (HelkaPL) [reply src/common/Section/styled.js:7]: Feel free to change it latter on witch next pull pequest :)

File: src/feautures/MovieList/MovieTile/styled.js (hunk 13 above)
Comment (HelkaPL) [reply src/feautures/MovieList/MovieTile/styled.js]: sure, let's change that latter, with next pull merge

File: src/feautures/MovieList/MovieTile/index.js (hunk 16 above)
Comment (HelkaPL) [reply src/feautures/MovieList/MovieTile/index.js:12]: sure, with next pull

File: src/feautures/MovieList/MovieTile/styled.js (hunk 17 above)
Comment (HelkaPL) [reply src/feautures/MovieList/MovieTile/styled.js:29]: we change it if it's gona be needed

File: src/feautures/MovieList/MovieTile/styled.js (hunk 23 above)
Comment (HelkaPL) [reply src/feautures/MovieList/MovieTile/styled.js:73]: sure, i will change it later when i check out yours code.

File: src/feautures/MovieList/styled.js (hunk 25 above)
Comment (HelkaPL) [reply src/feautures/MovieList/styled.js]: we checkout it later

**Comment by HelkaPL** [comment]
added mayor changes

File: src/common/Section/styled.js (hunk 8 above)
Comment (marcin10lw) [reply src/common/Section/styled.js:7]: Don't see the point of that. It's a quick change and it will speed up our work

**Review by marcin10lw — APPROVED** [review]

**Review by mikolajgraczyk — APPROVED** [review]
